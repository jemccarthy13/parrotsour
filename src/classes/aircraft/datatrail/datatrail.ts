import { Point } from "../../point"
import { IDMatrix } from "../id"

// TODO -- DATATRAIL -- register? so that Aircraft can generate one of each type of datatrail?
// TODO -- DATATRAIL -- Support a toggle for the different Sensor Types.....
export abstract class DataTrail {
  private startPos: Point

  constructor(startPos: Point) {
    this.startPos = startPos
  }

  public getStartPos(): Point {
    return this.startPos
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  public move(_heading: number): void {
    //console.warn("Moving DataTrail in base class. Will have no effect.")
  }

  abstract getCenterOfMass(heading: number): Point

  abstract draw(heading: number, id: IDMatrix): void
}
