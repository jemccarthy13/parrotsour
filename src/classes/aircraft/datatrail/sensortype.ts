export enum SensorType {
  RAW,
  ARROW,
}
