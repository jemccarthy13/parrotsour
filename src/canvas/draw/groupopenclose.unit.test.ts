import { SensorType } from "../../classes/aircraft/datatrail/sensortype"
import { AircraftGroup } from "../../classes/groups/group"
import { getOpenCloseAzimuth } from "./formatutils"
import { PaintBrush } from "./paintbrush"

// eslint-disable-next-line @typescript-eslint/no-var-requires
const canvasSerializer = require("jest-canvas-snapshot-serializer")
expect.addSnapshotSerializer(canvasSerializer)

describe("FormatUtils", () => {
  const canvas = document.createElement("canvas")
  canvas.width = 200
  canvas.height = 200
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  const ctx: CanvasRenderingContext2D = canvas.getContext("2d")!
  PaintBrush.use(ctx)
  PaintBrush.clearCanvas()

  describe("groupOpenClose_BlueInTheEast", () => {
    let grp1: AircraftGroup
    let grp2: AircraftGroup

    beforeEach(() => {
      grp1 = new AircraftGroup({
        sx: 50,
        sy: 25,
        hdg: 1,
        nContacts: 1,
      })

      grp2 = new AircraftGroup({
        sx: 50,
        sy: 100,
        hdg: 180,
        nContacts: 1,
      })

      PaintBrush.clearCanvas()
    })

    it("openclose_verywide_opening", () => {
      grp1.setHeading(1)
      grp2.setHeading(180)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" OPENING ")
    })

    it("openclose_onlyone_opening", () => {
      grp1.setHeading(90)
      grp2.setHeading(135)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" OPENING ")
    })

    it("openclose_onlyone_closing", () => {
      grp1.setHeading(90)
      grp2.setHeading(45)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" CLOSING ")
    })

    it("openclose_onlyone_sg_closing", () => {
      grp1.setHeading(135)
      grp2.setHeading(90)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" CLOSING ")
    })

    it("openclose_onlyone_ng_opening", () => {
      grp1.setHeading(45)
      grp2.setHeading(90)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" OPENING ")
    })

    it("openclose_narrow_closing", () => {
      grp1.setHeading(135)
      grp2.setHeading(45)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" CLOSING ")
    })

    it("openclose_normal_opening", () => {
      grp1.setHeading(45)
      grp2.setHeading(135)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" OPENING ")
    })

    it("openclose_westerly_closing", () => {
      grp1.setHeading(270)
      grp2.setHeading(45)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" CLOSING ")
    })

    it("openclose_west_closing", () => {
      grp1.setHeading(270)
      grp2.setHeading(330)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" CLOSING ")
    })

    it("openclose_west_opening", () => {
      grp1.setHeading(330)
      grp2.setHeading(270)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" OPENING ")
    })

    it("openclose_drastic_closing", () => {
      grp1.setHeading(180)
      grp2.setHeading(360)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" CLOSING ")
    })

    it("openclose_verynarrow_closing", () => {
      grp1.setHeading(92)
      grp2.setHeading(90)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_heading_too_close_together_closing", () => {
      grp1.setHeading(95)
      grp2.setHeading(85)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_heading_too_close_together_opening", () => {
      grp1.setHeading(275)
      grp2.setHeading(280)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_heading_west_closing", () => {
      grp1.setHeading(225)
      grp2.setHeading(295)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" CLOSING ")
    })

    it("openclose_heading_opposite_closing", () => {
      grp1.setHeading(225)
      grp2.setHeading(45)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" CLOSING ")
    })

    it("openclose_parallel", () => {
      grp1.setHeading(90)
      grp2.setHeading(90)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_parallel_west", () => {
      grp1.setHeading(270)
      grp2.setHeading(271)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })
    it("openclose_90_offset", () => {
      grp1.setHeading(1)
      grp2.setHeading(90)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" OPENING ")
    })
    it("openclose", () => {
      grp1.setHeading(310)
      grp2.setHeading(40)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_multicontact-closing-west", () => {
      grp1 = new AircraftGroup({
        sx: 50,
        sy: 25,
        hdg: 180,
        nContacts: 3,
      })
      grp2 = new AircraftGroup({
        sx: 50,
        sy: 150,
        hdg: 290,
        nContacts: 4,
      })
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" CLOSING ")
    })
  })

  describe("getGroupOpenClose_BlueInTheSOUTH", () => {
    let grp1: AircraftGroup
    let grp2: AircraftGroup

    beforeEach(() => {
      grp1 = new AircraftGroup({
        sx: 50,
        sy: 50,
        hdg: 1,
        nContacts: 1,
      })

      grp2 = new AircraftGroup({
        sx: 100,
        sy: 50,
        hdg: 180,
        nContacts: 1,
      })

      PaintBrush.clearCanvas()
    })

    it("openclose_verywide_opening", () => {
      grp1.setHeading(271)
      grp2.setHeading(90)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" OPENING ")
    })

    it("openclose_opp_leadtrail", () => {
      grp1.setHeading(1)
      grp2.setHeading(180)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_samedir", () => {
      grp1.setHeading(90)
      grp2.setHeading(135)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_onlyone_closing", () => {
      grp1.setHeading(90)
      grp2.setHeading(45)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_onlyone_sg_closing", () => {
      grp1.setHeading(135)
      grp2.setHeading(90)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_onlyone_ng_opening", () => {
      grp1.setHeading(45)
      grp2.setHeading(90)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_narrow_closing", () => {
      grp1.setHeading(135)
      grp2.setHeading(45)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_normal_opening", () => {
      grp1.setHeading(45)
      grp2.setHeading(135)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_westerly_closing", () => {
      grp1.setHeading(270)
      grp2.setHeading(45)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" OPENING ")
    })

    it("openclose_west_closing", () => {
      grp1.setHeading(270)
      grp2.setHeading(330)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_west_opening", () => {
      grp1.setHeading(330)
      grp2.setHeading(270)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_drastic_closing", () => {
      grp1.setHeading(180)
      grp2.setHeading(360)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_verynarrow_closing", () => {
      grp1.setHeading(92)
      grp2.setHeading(90)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_heading_too_close_together_closing", () => {
      grp1.setHeading(95)
      grp2.setHeading(85)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_heading_too_close_together_opening", () => {
      grp1.setHeading(275)
      grp2.setHeading(280)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_heading_west_closing", () => {
      grp1.setHeading(225)
      grp2.setHeading(295)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_heading_opposite_closing", () => {
      grp1.setHeading(225)
      grp2.setHeading(45)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" OPENING ")
    })

    it("openclose_parallel", () => {
      grp1.setHeading(90)
      grp2.setHeading(90)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })

    it("openclose_parallel_west", () => {
      grp1.setHeading(270)
      grp2.setHeading(271)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual("")
    })
    it("openclose_90_offset", () => {
      grp1.setHeading(1)
      grp2.setHeading(90)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" OPENING ")
    })
    it("openclose", () => {
      grp1.setHeading(310)
      grp2.setHeading(40)
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" OPENING ")
    })

    it("openclose_multicontact-closing-west", () => {
      grp1 = new AircraftGroup({
        sx: 50,
        sy: 25,
        hdg: 180,
        nContacts: 3,
      })
      grp2 = new AircraftGroup({
        sx: 50,
        sy: 150,
        hdg: 290,
        nContacts: 4,
      })
      grp1.draw(SensorType.ARROW)
      grp2.draw(SensorType.ARROW)
      expect(canvas).toMatchSnapshot()
      expect(getOpenCloseAzimuth(grp1, grp2)).toEqual(" CLOSING ")
    })
  })
})
