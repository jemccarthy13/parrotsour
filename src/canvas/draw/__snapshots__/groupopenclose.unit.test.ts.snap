// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose 1`] = `
<canvas
  data-snapshot-image="0195d07f219d06c180bc0e6436a8af640728db98eeddc2087acbc5de98080cc48ba58f01a82d99101d286d17df05020d3a34eb07a61007adfd7cff6df35b854f"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_90_offset 1`] = `
<canvas
  data-snapshot-image="4044c30de07a334bb2c6f1dbbe883868c25c274070ac936190ab307f715284ede68e10dfb8218500fe6d8c7e8df8a015767cfaa1efb029bf3996b9f7a62b28c0"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_drastic_closing 1`] = `
<canvas
  data-snapshot-image="50d4e6abeb8808b344916477dd340d3dca5c8687757a2a998a9e76127c15c47462ac22441ec094868c64053170df12f089235d81b1e17e539b15cb2c5327749f"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_heading_opposite_closing 1`] = `
<canvas
  data-snapshot-image="766489a7180339cf0bb14086942ea35f916bb07f33ed78a97838539eddab820357c11b477c58c68eeb0c6167e43c6ed6e275c58c0b7f90f8552d16ad25dbe38a"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_heading_too_close_together_closing 1`] = `
<canvas
  data-snapshot-image="0c06d79f2b51f89b8a7712073097ed982d78df48cdcac60f8f420cae9efb5779152b789e2c6d2d8ef15cc2f6d7ace469fff1e4c6e8c326af98c29fd39c2dff8c"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_heading_too_close_together_opening 1`] = `
<canvas
  data-snapshot-image="ff40b4aa44e0b541e065a9354ac655dcecb9b7886e6d6b226b76b67ccda0218b446729c7f08d3d98cd66ab52c9e747f67bcae233bf9533909c0c08939a916899"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_heading_west_closing 1`] = `
<canvas
  data-snapshot-image="f976d71b894dcf797b06249c65c2ae4aa1cb27e8b29ac3b9badde3298b276b81b72985f428f07b8b2446b0813dc093e970f57b0fb67290139b2f6003bbd19076"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_multicontact-closing-west 1`] = `
<canvas
  data-snapshot-image="2c212c7a52277bddafb07ac0d82ca6e0b71009af9fef357763c2a28305323e4037882b9ccb09ec147ac098a9b82933407db7fcf590ff1e5191deb345c6d8a8b2"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_narrow_closing 1`] = `
<canvas
  data-snapshot-image="a4d9544aaaa932db99da93ea0d8b769b83e3f100fada7e172fcca6ba57bf8d31d04c8dda88cc0346263bf91203f04094680ab59babf5cf782949bfed0aedbc94"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_normal_opening 1`] = `
<canvas
  data-snapshot-image="14b0af734e02d9eabedc8e9abc582b85cb287abec55e1d230d6d71445ce14e12d5461bc80cc3095d09cae119d81595215ac30985d88cc9760725a78782d1ba8d"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_onlyone_closing 1`] = `
<canvas
  data-snapshot-image="5133950052be3f75081c6375551f4600a04d62164b36d89a58a9d0592f37578ef38462f8ff0a440ac1e45492852c4ff3717b8af9a052285b2d1a7eaf463d5ea2"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_onlyone_ng_opening 1`] = `
<canvas
  data-snapshot-image="aff740988e86841da6fd9732047a24a358d40550ae57ac2bf87c3f2e461d3651879f67b15ab8c88d9eeb77695f595b34f8dbe5e2361f2c255c58ac85b06d2dbb"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_onlyone_sg_closing 1`] = `
<canvas
  data-snapshot-image="edb6cf832ece9c2d96ce78b4896a0eb63a45dd994175a846080956af64386cf9d288c24a4f1f5a5236e513191a2e5e04542bc2e3810215023dc8953cdf025c30"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_opp_leadtrail 1`] = `
<canvas
  data-snapshot-image="37d00d3a0249d9d31731ab4a51b144ab252434a572da6bb6be8efe01ba053a7216c0edfe839bf73678e810389dbbdb8ef303de02e129d94933f8b5293890bb23"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_parallel 1`] = `
<canvas
  data-snapshot-image="e537adecdaf4a949c3d3ce4cb2674bdc98662a1ab2902e88a3ce2f1f6ebe8c0359d4a504f6e6bde6b141b6196447ff90e006b5df095f713270fa2a0ff3923ec7"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_parallel_west 1`] = `
<canvas
  data-snapshot-image="1649a8a0762c2f7d6b4f401555cc0d0bc9cd21fa48f1f9963bc33c1d25fdebe0292e488a5b745a418c59731824861106f10090712eeb38c56b49ed8a5679d27f"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_samedir 1`] = `
<canvas
  data-snapshot-image="850245b5ef9b932dacf7df9bca50282119c381c823c5fe43edf50f61412601eab285ed27c814308083e8993d607f59c7e807ded67a04fe7ef5e8d142bd7b5e6c"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_verynarrow_closing 1`] = `
<canvas
  data-snapshot-image="31e9051870bfb5aa1c4a54ab9bdb59c1301abba63f3e50670f9e02d1d32ccdf81d376a635e85795e99658e26285777b137580e6dd598947e7445df961928c39b"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_verywide_opening 1`] = `
<canvas
  data-snapshot-image="c0ead56e358b5819f4ebc5746ae32eaafdc0044fa73e81127babc7e7f3ddc120b242c166a50b7d2d3f6077e867659124902cd555aeb7efcd551442f5bb135aec"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_west_closing 1`] = `
<canvas
  data-snapshot-image="99acf315cf2770e0592eb46e794d82a3a55ba05791af9715e769983285fb9989fa45aeb3894a9cf3650ea999c78851c5fabce31f9eac3ae89aa8ba356b465ab0"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_west_opening 1`] = `
<canvas
  data-snapshot-image="cfce6ea23ee6df5dd5628861c8f7fb19e02ffa01b3ec84e137dc6151985931cd591e6e7f7cd845773d60b5e2e2e79ea78e42bc48a33a84389c63050879f784fa"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils getGroupOpenClose_BlueInTheSOUTH openclose_westerly_closing 1`] = `
<canvas
  data-snapshot-image="4c94261957bcc275cbfeffcfdbe1662821fb2d6d7eee3f1d234a3d6d6cd7fbf5438a7a64096bf3b863c931f6e9cbac59ed619b410910946c26de826113d81569"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose 1`] = `
<canvas
  data-snapshot-image="444cda0e99bf9bb0425d8c577f28ab32f2632b5e82784ebcdbe78c9bd436c94734b86df9d5f85a0933f7427f6e33f6af02ec9b7ccf4349dd9037c7e463613889"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_90_offset 1`] = `
<canvas
  data-snapshot-image="baa9c8c8c322439f1e37bf93649370e93d61263a162b49825761e0d79cbaea25592e5dc74eca545c44c474a48e384dac6f172768bb08a0b02cb82344b8461ab8"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_drastic_closing 1`] = `
<canvas
  data-snapshot-image="02e8b384279997a6b2a7caa4a030a29be0781a84722699fc94ea4abe65335814674654ea80e8a7ca3526b8958a4efb3dd625e788b8b2394e2d31a98b0470ca78"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_heading_opposite_closing 1`] = `
<canvas
  data-snapshot-image="9ddad500a32b59e936242b285c6e49d2cec993ea0db1236415b685675340c45fbae026124eb70c14e4888ab03d7a4fe43be41c49a819a03c82e5c994f6b2633b"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_heading_too_close_together_closing 1`] = `
<canvas
  data-snapshot-image="531265f740878efb4da500388668ba2e6b177c6f0d98b4fce0219691ab2c2cbb78844c57491cb83ec12297fcd70fdaeeb1f3d341260dd05b84651f45ba157000"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_heading_too_close_together_opening 1`] = `
<canvas
  data-snapshot-image="ddde1cacb52f7259153ec10a11dc12fb28ab2c13a812860ef677c3d46e293bc6459a076de196ea274d93075c594901100b6de117de5656f627eb862c389f48fc"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_heading_west_closing 1`] = `
<canvas
  data-snapshot-image="09aa735055aa15d2ba80cc6c399e69b7a04ca5d8dab832c7c8c74fac743d6f56b5d4d94299df50b069e132f22548bdc5ec319da4cf54d47066fe2288d29cba4d"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_multicontact-closing-west 1`] = `
<canvas
  data-snapshot-image="2c212c7a52277bddafb07ac0d82ca6e0b71009af9fef357763c2a28305323e4037882b9ccb09ec147ac098a9b82933407db7fcf590ff1e5191deb345c6d8a8b2"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_narrow_closing 1`] = `
<canvas
  data-snapshot-image="eac5e40532db4942d03deba9b087718e43e2063ecb0e85c27c5e7ebed3eee72cc62e91e1a17595f25b828d4e69ef9514dc4bd758fed6cc9e854f37a880878c7b"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_normal_opening 1`] = `
<canvas
  data-snapshot-image="4f88e7cece724cf6dacad290c7c6c57b5c395cb0dbfd6cc8e417e004dc97284e2c8927bff5297aa567303fbae622cda4200d589f731ceb3aae2208a09aa4c7ff"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_onlyone_closing 1`] = `
<canvas
  data-snapshot-image="5648493bc44ba377a21ee4cc67eaf49db65d4eaad7fdb86975e0c081672497912ccb5f2a4713ab848ba6958a61b384a503cde4cb08529bcfda2fea82879e9c7a"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_onlyone_ng_opening 1`] = `
<canvas
  data-snapshot-image="93509f5f46208a7d0738fa9331a4d4d87bcb58a06e45c706e084ddcff7322269dc852b61b31d4855be0a4863214374f55ea30206ee1b0d1bfd7ab0e4bc29fb17"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_onlyone_opening 1`] = `
<canvas
  data-snapshot-image="9db3903c3a1cd19962f18497a3f6d8d807449da21a34d6ec4c6e182af2acb7d4fcabb5a6dd27c64f13adbb41292137e20cfab7dd31d024454afca2728c44f81f"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_onlyone_sg_closing 1`] = `
<canvas
  data-snapshot-image="54bd7517e3aac141b46570cae6c9bbf87631bff79fa9524be86cecf435aae684b30d64fe0c23430c2a08457898d47366c843dfba83c8efe1a0a60ca4d30feb9b"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_parallel 1`] = `
<canvas
  data-snapshot-image="5ddac949811988265a11246bee3566ec5d7d14ac798ba925dc0edb435f20a0d17634e20bcae6aa446c3717b50a44984cd063b88f4212c21f575fc526300eede9"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_parallel_west 1`] = `
<canvas
  data-snapshot-image="00abced6790bb8a50dec4bbb467df051a45291877bf1c259bc80343d6100bb7fe92eb8d8ac23f90cfef628661d07431620ac26ef8de32b89c4e2cc75f9fbe1ba"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_verynarrow_closing 1`] = `
<canvas
  data-snapshot-image="bffc31fc8fcb6b5ea3ee15f55f66b651cdf3dfb7dc0ff9c3e141cc931dbdceef9cba47daad6ce89895bcb1b713ceb87bec8d66904cc44fb1f31fbbda4a692e70"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_verywide_opening 1`] = `
<canvas
  data-snapshot-image="a6024cec343d87c40754c459a18837b53a2906c0c27d0684da164e6f108f86e38acef1ed32c6c8af055bd10fb785da3b62d411fdb8cd2c0aa7eedb36eb590a90"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_west_closing 1`] = `
<canvas
  data-snapshot-image="0fb08be539ffb498b9e5292c753fd17a4764309b5af4ae5723b062b41a2f64093a583f692a294beb23b91cebfc79920a81b29de3453c8a94a0b93ff806c9b28b"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_west_opening 1`] = `
<canvas
  data-snapshot-image="e8490e10c1ef54edd887cb197c945242881e3994f3b1dcfc9407d2ae864f544339e6c65c1683dc4df2c6fc496bdf341f019b0d1656d93fadaf9b83a4c6215d3f"
  height="200"
  width="200"
/>
`;

exports[`FormatUtils groupOpenClose_BlueInTheEast openclose_westerly_closing 1`] = `
<canvas
  data-snapshot-image="d7ee851d4dee4bc2391d684e7dc4a761de88ed252884d317a87922861b7039de8dd3b8aa1c20d5291eef258906ca747b35e6e7cf7775be1b8d35e7a727810a4e"
  height="200"
  width="200"
/>
`;
